test/pools/unit/types/ShareClassId.t.sol

```diff
@@ -21,6 +21,18 @@ contract ShareClassIdTest is Test {
         PoolId poolId = PoolId.wrap(poolId_);
         ShareClassId scId = newShareClassId(poolId, index);
 
-        assertEq(scId.raw(), bytes16(uint128(poolId.raw()) + index));
+        assertEq(scId.raw(), bytes16(uint128(poolId.raw() << 64) + index));
+    }
+
+    function testShareClassIdCollosionResistance(uint64 poolId, uint32 index) public pure {
```

Comment:
Wdyt about adding a second `poolId` and `index` input, and ensure that no share class ID of `poolId1` can match no share class ID of `poolId2`?